tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -427,7 +427,12 @@ func (iptConfigurator *IptablesConfigurator) run() {
 			"-p", "udp", "--dport", "53",
 			"-j", "REDIRECT", "--to-port", "15013")
 	}
-
+	if iptConfigurator.cfg.InboundInterceptionMode == constants.TPROXY && tproxyPreserveSourceIP.Get() {
```

Comment:
Do this unconditionally.

Reply:
Thanks for the review, @rlenglet
I agree with this suggestion and will take a modification tomorrow. by the way, configuration for original_src also committed, PTAL.